### PR TITLE
Don't hover outside data in tree

### PR DESCRIPTION
**Pull request by andydotxyz**
Just don't show hover marks when cursor is outside data.

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: widget/table_test.go [hunk 1]
```diff
@@ -117,6 +117,25 @@ func TestTable_Hovered(t *testing.T) {
 	defer w.Close()
 	w.Resize(fyne.NewSize(180, 180))
 
+	test.MoveMouse(w.Canvas(), fyne.NewPos(35, 100))
+
+	assert.Equal(t, -1, table.hoveredColumn)
+	assert.Equal(t, -1, table.hoveredRow)
+
+	test.AssertImageMatches(t, "table/nohovered.png", w.Canvas().Capture())
```
Comment (stuartmscott) [inline widget/table_test.go:125]: nit: no_hover or not_hovered

File: widget/table.go [hunk 2]
```diff
@@ -360,13 +360,22 @@ func (c *tableCells) hoverAt(pos fyne.Position) {
...
+	}
+
 	if c.t.moveCallback != nil {
 		c.t.moveCallback()
 	}
 }
 
 func (c *tableCells) hoverOut() {
-	c.t.hoveredRow = -1
+	c.t.hoveredColumn = -1
```
Comment (stuartmscott) [inline widget/table.go:378]: Wow how'd I miss that? Did the test catch it?

File: widget/table_test.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ func TestTable_Hovered(t *testing.T) {
 	app.Settings().SetTheme(theme.LightTheme())
 
 	table := NewTable(
-		func() (int, int) { return 3, 5 },
+		func() (int, int) { return 2, 2 },
```
Comment (stuartmscott) [inline widget/table_test.go:107]: What caused this change, was there too many cells for the canvas size, or is there something special about square tables?

File: widget/table_test.go [hunk 4]
```diff
@@ -117,6 +117,25 @@ func TestTable_Hovered(t *testing.T) {
...
+	test.AssertImageMatches(t, "table/nohovered.png", w.Canvas().Capture())
+
+	table = NewTable(
+		func() (int, int) { return 3, 5 },
+		func() fyne.CanvasObject {
+			return NewLabel("placeholder")
+		},
+		func(row, col int, c fyne.CanvasObject) {
+			text := fmt.Sprintf("Cell %d, %d", row, col)
+			c.(*Label).SetText(text)
```
Comment (stuartmscott) [inline widget/table_test.go:134]: nit: could be one line
```suggestion
			c.(*Label).SetText(fmt.Sprintf("Cell %d, %d", row, col))
```

**Review by stuartmscott — CHANGES_REQUESTED** [review]
Couple small things, otherwise looks good!

File: widget/table_test.go (hunk 3 above)
Comment (andydotxyz) [reply widget/table_test.go:107]: It is not a change - this was inserted, I think you are seeing an artifact of how github presents the info

Comment (andydotxyz) [reply widget/table_test.go:107]: The additional table (first one) is smaller so we can mouse out of it in the code below

Comment (stuartmscott) [reply widget/table_test.go:107]: Ahh I see

**Review by stuartmscott — APPROVED** [review]

**Pull request merged**